### ISSUE TITLE
Tab-completed paths aren't highlighted

### ISSUE THREAD

**Issue by rjww** [zsh-users/zsh-syntax-highlighting#519]
I have distinct highlighting set for path separators. When tab-completing a path, this highlighting doesn't appear. So, if I manually type
```
cd ~/Dropbox
```
then highlighting will show up as expected. But if I then press `<TAB>`
```
cd ~/Dropbox/
```
the path highlighting disappears. At first I thought this was an issue with the trailing `/`, but if, for example, I press `<Ctrl-L>` to clear the terminal _after_ tab-completing the above path, syntax highlighting on the path will reappear (including on the trailing slash). So perhaps the problem is that `<TAB>` isn't calling the highlight function?

**Comment by danielshahaf** [comment]
> I have distinct highlighting set for path separators. When tab-completing a path, this highlighting doesn't appear. So, if I manually type
> ```
> cd ~/Dropbox
> ```
> then highlighting will show up as expected. But if I then press `<TAB>`
> ```
> cd ~/Dropbox/
> ```
> the path highlighting disappears. At first I thought this was an issue with the trailing `/`, but if, for example, I press `<Ctrl-L>` to clear the terminal _after_ tab-completing the above path, syntax highlighting on the path will reappear (including on the trailing slash). So perhaps the problem is that `<TAB>` isn't calling the highlight function?

Is z-sy-h loaded last in your zshrc?

What's the output of `bindkey $'\t'` and of `print -r -- $widgets[${"$(bindkey $'\t')"#* }]`?

**Comment by sipior** [comment]
I see the same problem here. This is with a completely empty .zshrc, save for loading antigen and the requisite "antigen bundle" statement for z-sy-h. In addition I've noticed that, when I also turn on zsh-autopair (again, only the two plugins are present in .zshrc), when opening a new pair of quotes or braces, the entries are not highlighted, but are immediately highlighted when text is entered in between the matched pair, or the cursor moves in any way.

To answer the question you posed to rjww: 

%  bindkey $'\t'  
"^I" expand-or-complete
% print -r -- $widgets[${"$(bindkey $'\t')"#* }]
completion:.expand-or-complete:_main_complete

This is with only z-sy-h loaded.

I can confirm that the problem does *not* appear in version 0.6.0-rc2-dev; I had updated to 0.7.0-dev yesterday, and noticed it directly.

Anyway, a very minor cosmetic problem, but I guess it bothered me enough to head to github to see if someone else had noticed it, or to see if I perhaps was going crazy :-)

**Comment by danielshahaf** [comment]
> % print -r -- $widgets[${"$(bindkey $'\t')"#* }]
> completion:.expand-or-complete:_main_complete

This means the `expand-or-complete` widget isn't z-sy-h-wrapped.  That's not exactly surprising, but now we know what the immediate problem is.  The expected output is something like `user:_zsh_highlight_widget_orig-s0.0000030000-r28743-expand-or-complete` (the numbers vary).

Could you bisect?

You may also try the `feature/redrawhook` branch, which is supposed to be immune to this kind of issue (sorry, forgot to mention it earlier).

**Comment by sipior** [comment]
Well, I had a go at bisecting, with amusing results. First: the feature/redrawhook branch worked flawlessly; many thanks for the pointer.

I couldn't replicate the "good" result anymore, even with version 0.6.0-rc2-dev (when used with antigen). I noticed, however, that sourcing  zsh-syntax-highlighting.zsh manually would fix the problem immediately. This was even true with an empty .zshrc (save for the antigen bundle statement for z-sy-h): the problem persisted, but sourcing the plugin file (again) fixed it. I recently moved to antigen to manage the few plugins I do use; I had assumed that it, ultimately, simply sourced the plugin scripts in order. In any event, adding an extra "source zsh-syntax-highlighting.plugin.zsh" at the end of my .zshrc has things working as well as before. Ironically, I moved to antigen to make my life simpler.

I'm a little puzzled. In any event, until I can figure out exactly what antigen is doing, it seems reasonable to conclude that the problem does not lie with z-sy-h itself; apologies for wasting your time with this. Thanks again!
